packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -740,6 +752,110 @@ const Conversation = WebexPlugin.extend({
+
+    return response.body;
+  },
+
+  /**
+   * Return a list of child activities with a given conversation, parentId and other constraints.
+   * @param {string} conversationUrl targeted conversation URL
+   * @param {string} activityParentId parent id of edit activities or thread activities
+   * @param {object} query object containing query string values to be appended to the url
+   * @param {string} query.activityType The type of child activities to return.
+   * The value is one of 'reply', 'edit', 'cardAction', 'reaction', 'reactionSummary', 'reactionSelfSummary'
+   * @param {number} [query.limit] The limit of child activities that can be returned per request
+   * @param {boolean} [query.latestActivityFirst] Sort order for the child activities
+   * @param {boolean} [query.includeParentActivity] Enables the parent activity to be returned in the activity list
+   * @param {string} [sinceDate] Get all child activities after this date
+   * @param {string} [maxDate] Get all child activities before this date
+   * @returns {Promise<Array>}
+   */
+  async listChildActivitiesByParentId(conversationUrl, activityParentId, query) {
+    const params = {
+      method: 'GET',
+      url: `${conversationUrl}/parents/${activityParentId}`,
+      qs: query
+    };
+
+    return this.request(params);
+  },
+
+  /**
+   * Returns a list of _all_ child activities for a given parentId within a given conversation
+   * @param {object} options
+   * @param {string} [options.conversationUrl] targeted conversation URL
+   * @param {string} [activityParentId] parent id of edit activities or thread activities
+   * @param {object} [query] object containing query string values to be appended to the url
+   * @param {string} [query.activityType] The type of child activities to return.
+   * The value is one of 'reply', 'edit', 'cardAction', 'reaction', 'reactionSummary', 'reactionSelfSummary'
+   * @param {number} [query.limit] The limit of child activities that can be returned per request
+   * @param {boolean} [query.latestActivityFirst] Sort order for the child activities
+   * @param {boolean} [query.includeParentActivity] Enables the parent activity to be returned in the activity list
+   * @param {string} [sinceDate] Get all child activities after this date
+   * @param {string} [maxDate] Get all child activities before this date
+   * @returns {Promise<Array>}
+   */
+  async listAllChildActivitiesByParentId(options = {}) {
+    const {conversationUrl, activityParentId, query} = options;
+
+    const initialResponse = await this.listChildActivitiesByParentId(conversationUrl, activityParentId, query);
+
+    let page = new Page(initialResponse, this.webex);
+
+    const items = [...page.items];
+
+    while (page.hasNext()) {
+      // eslint-disable-next-line no-await-in-loop
+      page = await page.next();
+      for (const activity of page) {
+        items.push(activity);
+      }
+    }
+
+    if (last(items).published < items[0].published) {
```

Comment:
When would be the case that we don't want to reserve? Should we set the sort order in the request instead?

Reply:
I'm not sure, actually. I saw this condition in the `_list` method, and I assumed it had something to do with convo. I see @mwegman gitblame there in the _list method, maybe he can lend some insight on whether this is necessary check?